### PR TITLE
First attempt to modularize and restructure code

### PR DESCRIPTION
**Pull request by Datenlord1510**
**Changelog**
- Modularized code and wrote appropriate classes to encapsulate functionality
- Uppercased .py-files (e.g. cache.py -> Cache.py)
- Added doc-string to start of every file
- Adjusted typing hints and removed unnecessary ones (_main should never return anything_)
- Adjusted codebase according to the PEP8 styleguide with minor exceptions for readabilty
- removed generation.py
- removed web.py

**Config.py**
- A class to compile configuration settings in a single instance
- within the file is the constant CONFIG, which is a Config instance
- CONFIG.CONFIG is used to access configuration that is not directly an attribute like DATA_DIR

**APIHelper.py**
- Inherited by other classes to avoid code duplication.
- Used to send requests or websearch

**Cache.py**
- A class to handle database actions.
- Added a table for "processed" instead of using another file.
- get_response_by_url is not used but was left for eventual later usage
- insert_response is not used but was left for eventual later usage
- As of now the requests table is not used.
  Only processed is used as a cache, consisting of dirnames.

**Generator.py**
- wraps the functions that were in generation.py 

**PREdbs.py**
- used to retrieve information from PREdbs

**Response.py**
- a custom class mirroring requests.Response

**Pre.py**
- Just a class representing a PRE
- Was put into its own class to make import easier and maybe extend it later on

**Steam.py**
- Used to handle the Steam store

**GOG.py**
- Used to handle the GOG store

**Epic.py**
- Used to handle the Epic store

**StoreHandler.py**
- Wrapper for all the stores.
- An instance of each store is set as an attribute so you can easily access them like StoreHandler().steam.method()

### CODE REVIEW

File: dailyreleases/stores/epic.py [hunk 1]
```diff
@@ -1,48 +1,50 @@
...
-@retry()
-def get_epic_games_data(query: str):
-    search_json = api.fetch_store_games(keywords=query)
-    return search_json
+    @retry()
+    def load_offerid_json(self):
+        egs_offerid_url = CONFIG.CONFIG['main']['egs_offeridapi_url']
+        logger.debug("Loading EGS offerid defintions from " + egs_offerid_url)
+        offerid_txt = requests.get(egs_offerid_url).content.decode()
+        offerid_json = loads(offerid_txt)
```
Comment (amir16yp) [inline dailyreleases/stores/epic.py]: should be `self.offerid_json`

Comment (Datenlord1510) [reply dailyreleases/stores/epic.py]: You are right, just fixed it. Thanks!

**Comment by Datenlord1510** [comment]
Final commit should be come the next days. There's a weird bug right now where it retrieves NSW releases too.

**Comment by Datenlord1510** [comment]
As of now, the Pres from the current day and every Pre from yesterday that is **not** stored in the database is returned and formatted into a post. This means the package should be run in debug mode one time before its deployed to production.

File: dailyreleases/main.py [hunk 2]
```diff
@@ -3,15 +3,19 @@
 from time import sleep
 from discord_webhook import DiscordWebhook
 import prawcore
-from .stores import epic
+
+from .stores import StoreHandler
```
Comment (amir16yp) [inline dailyreleases/main.py]: should be `from .stores import StoreHandler.StoreHandler`

Comment (amir16yp) [reply dailyreleases/main.py]: sorry i meant `from .stores.StoreHandler import StoreHandler`

**Comment by Datenlord1510** [comment]
Again regarding the time to collect and post releases:
As mentioned above, in case releases were missed a day before, they are posted too. If you want to make sure it's just the releases from the current day, I can change line 100 in Generator.py to:
 `relevant_pres = [pre for pre in pres if pre.from_today() is True]`.
I'm just worried that when the process starts at midnight, the routine tries to post the releases from the already new day.

**Comment by amir16yp** [comment]
sorry for the delay in merging. im back on working on this project

**Pull request merged**